runner.go

```diff
@@ -82,6 +82,45 @@ func ExecutorOptions(opts ...ExecutorOption) RunOption {
 	}
 }
 
+// groupTaskAndFlagArgs groups all tasks with their flag arguments.
+// Notably, this function does not group flags passed to taskrunner itself.
+// Those flags are stored via the flags package and are handled separately.
+func (r *Runtime) groupTaskAndFlagArgs(args []string) map[string][]string {
+	flagArgsPerTask := map[string][]string{}
+
+	var currTaskName string
+	var currFlagsList []string
+	for _, arg := range args {
+		// Check task registry to figure out whether arg the is a task or a flag
```

Comment:
possible edge case - is it possible to add a task with a name that has a `-` in front? 

should we validate that the arguments that are flags have a suffix of `-`?

Reply:
Hmm

Rn we do not validate whether task names follow any format, so it could be either prefixed or suffixed with a `-`

I think we will still catch those as "tasks" though since they'll be registered in the registry with their prefix/suffixes (and we check for whether it's at ask before assuming it's a flag)

We do eventually do validation in the executor though for `-`/`--` prefixes 👍